### DISCUSSION TITLE
Custom Workflow

### DISCUSSION THREAD

**Discussion by karkir0003** [community/community#45653] [Projects and Issues]
### Select Topic Area

General

### Body

When will users be able to specify custom workflows in the new version of Github Projects such that we can say something like "if an issue is marked with status "Done", Auto Archive it" or any sort of custom project board automations

**Comment by amichai-bd** [comment]
I have the same request as the original poster regarding custom workflows in GitHub Projects. 
In my case, I have multiple types of tasks that move through a typical "Todo -> In_progress -> Done" workflow, but with different names for each type of task.
![image](https://user-images.githubusercontent.com/81047407/221377012-7c2f8e78-6651-4a88-a6e7-ca4800ccc16e.png)


It would be incredibly helpful if there was a way to automate the movement of tasks from "In_progress \<name\>" to "Done \<name\>" **when the associated issue is closed.** 

![image](https://user-images.githubusercontent.com/81047407/221377173-592c7577-d804-4d33-b0ab-d5a3a35cbc3a.png)
![image](https://user-images.githubusercontent.com/81047407/221377190-cb60557d-0cea-478b-a84f-b609b05cc343.png)


Currently, I have to manually move tasks to the appropriate column in the project board, which can be time-consuming and error-prone.
I hope that GitHub will consider implementing this feature in the near future

**Reply by kjersbry** [reply]
I think you can solve this differently. Try creating a custom field instead of custom status values. For example a field "filter", with type "Single select". Then add options "BIG_CORE", "CACHE", etc. Then you don't need the custom status columns, you can just use the original "Todo" "In progress" "Done" etc. For each of the boards, use the filter "filter: "BIG_CORE"", instead of "status: "Todo BIG_CORE", "In progress BIG_CORE", ...".

**Comment by Tofull** [comment]
Could be awesome if the customization workflow can trigger a kind of github action. 
For instance, it would allow to easily send notification via webhook, email or anything else depending on the task status.

**Reply by karkir0003** [reply]
that would be amazing!

**Comment by jeherve** [comment]
Related discussions:

- https://github.com/orgs/community/discussions/55096
- https://github.com/orgs/community/discussions/53705

**Comment by probitcarwyn** [comment]
This is on the public roadmap here:

- https://github.com/github/roadmap/issues/281

**Comment by kjersbry** [comment]
I support the idea of custom workflows. In my case, I need a modified version of the "Item closed" workflow. I want to automatically move things from to one column to another when closed, but _only_ if they have a specific status.

**Comment by aarongable** [comment]
The public roadmap issue for this feature has been closed as "outdated", and no replacement issue has been added to the roadmap. Can we get any confirmation that this feature is even being considered? It is, after all, still prominently advertised on [the GitHub Issues and Projects homepage](https://github.com/features/issues):
<img width="1285" height="773" alt="image" src="https://github.com/user-attachments/assets/70c39cbe-2d36-49c5-89b9-a2cde4366863" />
